ssl/quic/quic_channel.c

```diff
@@ -1879,8 +1881,12 @@ static void ch_rx_handle_packet(QUIC_CHANNEL *ch)
      */
     if (!ch->is_server
         && ch->qrx_pkt->peer != NULL
-        && (BIO_ADDR_family(&ch->cur_peer_addr) == AF_INET
-            || BIO_ADDR_family(&ch->cur_peer_addr) == AF_INET6)
+        && (
+               BIO_ADDR_family(&ch->cur_peer_addr) == AF_INET
+#if OPENSSL_USE_IPV6
+            || BIO_ADDR_family(&ch->cur_peer_addr) == AF_INET6
+#endif
+        )
```

Comment:
This seemed to be the cleanest way to do this - it could be something like

```
    if (!ch->is_server
        && ch->qrx_pkt->peer != NULL
        && (BIO_ADDR_family(&ch->cur_peer_addr) == AF_INET
#if OPENSSL_USE_IPV6
            || BIO_ADDR_family(&ch->cur_peer_addr) == AF_INET6)
#else
           )
#endif
        && !bio_addr_eq(ch->qrx_pkt->peer, &ch->cur_peer_addr))
        return;
```

but that's even worse!